File: src/api/tauri/elevation.ts
/**
 * Elevation status IPC (Windows UAC / admin).
 */

import { invoke } from '@tauri-apps/api/core';

export interface ElevationStatus {
  elevated: boolean;
  autoElevate: boolean;
  platform: string;
}

export const getElevationStatus = (): Promise<ElevationStatus> =>
  invoke<ElevationStatus>('get_elevation_status');

/** Request UAC relaunch. Current process may exit when relaunch succeeds. */
export const requestElevation = (): Promise<ElevationStatus> =>
  invoke<ElevationStatus>('request_elevation');

File: src-tauri/src/commands/elevation.rs
//! Elevation status IPC.

use crate::elevation;
use crate::error::CoreError;

/// Returns whether the app is running elevated and whether auto-elevate is on.
#[tauri::command]
pub fn get_elevation_status() -> Result<serde_json::Value, CoreError> {
    Ok(elevation::status_json())
}

/// Explicitly request elevation (relaunch with UAC). Works in dev and release.
/// If relaunch succeeds this process exits; otherwise returns current status.
#[tauri::command]
pub fn request_elevation() -> Result<serde_json::Value, CoreError> {
    let _ = elevation::request_elevation_relaunch();
    Ok(elevation::status_json())
}
